src/Relation/Binary/Construct/Closure/Reflexive/Properties/WithK.agda

```diff
@@ -0,0 +1,72 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of reflexive closures which rely on the K rule
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe #-}
+
+module Relation.Binary.Construct.Closure.Reflexive.Properties.WithK where
+
+open import Data.Empty.Irrelevant using (⊥-elim)
+open import Data.Sum as Sum
+open import Relation.Binary
+open import Relation.Binary.Construct.Closure.Reflexive
+open import Relation.Binary.Construct.Closure.Reflexive.Properties public
+import Relation.Binary.PropositionalEquality as PropEq
+open PropEq using (_≡_; refl)
+
+module _ {a ℓ} {A : Set a} {_~_ : Rel A ℓ} where
+    irrel∧irrefl⟶irrel : Irrelevant _~_ → Irreflexive _≡_ _~_ → Irrelevant (Refl _~_)
```

Comment:
Looking at other modules in `Relation.Binary.Construct` it looks like we don't typically put the preconditions in the type, and simply write `irrel` and `antisym` etc.

Reply:
Done — was following `Relation.Binary.Consequences`